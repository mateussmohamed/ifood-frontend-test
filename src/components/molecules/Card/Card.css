.card {
  display: flex;
  justify-content: start;
  flex-direction: column;
  width: 250;
  max-width: 250;
  margin-right: 23px;
}
