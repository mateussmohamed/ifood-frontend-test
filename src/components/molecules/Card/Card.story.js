import { storiesOf } from '@storybook/react';
import React from 'react';
import Card from './index';

const stories = storiesOf('Card', module);

const props = {
  title: 'Bailando',
  image: 'https://i.scdn.co/image/723b3060add0aff0011fd2ae7d59237bbe788bb2',
  songs: 60,
};
stories.add('without props', () => (
  <div style={{ width: '100%' }}>
    <Card {...props} />
  </div>
));
