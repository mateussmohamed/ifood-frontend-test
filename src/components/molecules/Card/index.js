import React from 'react';
import PropTypes from 'prop-types';

import CardImage from '../../atoms/CardImage';
import CardTitle from '../../atoms/CardTitle';
import CardSongs from '../../atoms/CardSongs';

import './Card.css';

const Card = ({
  image,
  title,
  songs,
  action,
}) => (
  <div className="card">
    <CardImage src={image} title={title} />
    <a href="#!" role="button" onClick={action}>
      <CardTitle title={title} />
    </a>
    <CardSongs songs={songs} />
  </div>
);

Card.propTypes = {
  image: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  songs: PropTypes.number.isRequired,
  action: PropTypes.func.isRequired,
};

export default Card;
