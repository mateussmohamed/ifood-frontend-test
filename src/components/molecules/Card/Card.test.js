import React from 'react';
import renderer from 'react-test-renderer';
import Card from './index';

test('Should Card match to snapshot', () => {
  const props = {
    title: 'Bailando',
    image: 'https://i.scdn.co/image/723b3060add0aff0011fd2ae7d59237bbe788bb2',
    songs: 234
  };
  const component = renderer.create(<Card {...props} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
