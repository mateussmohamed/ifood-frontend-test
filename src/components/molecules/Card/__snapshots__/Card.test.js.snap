// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Card match to snapshot 1`] = `
<div
  className="card"
>
  <img
    alt="Bailando"
    className="card-image"
    src="https://i.scdn.co/image/723b3060add0aff0011fd2ae7d59237bbe788bb2"
    title="Bailando"
    width={250}
  />
  <a
    href="#!"
    onClick={[Function]}
    role="button"
  >
    <span
      className="card-title"
    >
      Bailando
    </span>
  </a>
  <span
    className="card-song"
  >
    234
     Songs
  </span>
</div>
`;
